src/dev/i18n/README.md

```diff
@@ -0,0 +1,208 @@
+
+  ```
+  const MyComponentContent = ({ intl }) => (
+    <input
+      type="text"
+      placeholder={intl.formatMessage({
+        id: 'plugin_namespace.message_id',
+        defaultMessage: 'Default message string literal',
+        values: { /*object*/ },
+        context: 'Message context'
+      })}
+    />
+  );
+  ```
+
+  `values` and `context` properties are optional.\
+  Callee of call expression should be either `intl.formatMessage` or `this.props.intl.formatMessage`.\
+  Expression can be parsed only if it is located in a JSX expression or in a block of JS code.
+
+#### JavaScript (primarily server-side) (.js, .ts, .jsx, .tsx)
+
+```js
+intl(
+  'plugin_namespace.message_id',
+  {
+    values: { /*object*/ },
+    defaultMessage: 'Default message string literal',
+    context: 'Message context'
+  },
+);
+```
+
+or
+
+```js
+intl.translate(
+  'plugin_namespace.message_id',
+  {
+    values: { /*object*/ },
+    defaultMessage: 'Default message string literal',
+    context: 'Message context'
+  },
+);
+```
+
+`values` and `context` properties are optional.\
+ Expression can be parsed only if it is located in syntactically valid JS/TS code. Do not use type assertions in TypeScript for `defaultMessage` or `context` properties, id argument or the second argument of `intl*` call expression. It is never needed for i18n engine use cases.
+
+#### Jade (.jade)
+
+```
+#{i18n('plugin_namespace.message_id', {
+  values: { /*object*/ },
+  defaultMessage: 'Default message string literal',
+  context: 'Message context',
+})}
+```
+
+`values` and `context` properties are optional.\
+Expression in `#{...}` is parsed as a JS expression.
+
+#### Handlebars (.handlebars, .hbs)
+
+```
+{{i18n 'plugin_namespace.message_id' '{"defaultMessage": "Default message string literal", "context": "Message context"}'}}
+```
+
+`values` and `context` properties are optional.`
+The third token (the second argument of i18n function call) should be a string literal that contains a valid JSON.
+
+### Usage
+
+```
+node scripts/extract_default_translations path/to/plugin path/to/another/plugin ...
+```
+
+`path/to/plugin` is a path to a root directory where messages searching should start, `en.json` will be created in `path/to/plugin/translations`.
+In case of parsing issues, exception with the necessary information will be thrown to console and extraction will be aborted.
+
+### Output
+
+The tool generates a partly plain JSON5 file. It contains injected `formats` object and plain structured `id: message` pairs.\
+If `context` is provided for a message, then its value will be written to result JSON as a comment after `id: message, \\ `.\
+Messages are sorted by id, but `formats` object is always on top of JSON.
+
+**Example**:
+```js
+{
+  formats: {
+    ...
+  },
+  'plugin_namespace.message.id-1': 'Default message text 1',
+  'plugin_namespace.message.id-2': 'Default message text 2', // Message context
+}
+```
+
+## Locale files verification tool
```

Comment:
As mentioned higher, I would suggest to create folder for each tool and use `README` per tool
What do you think?
@azasypkin @LeanidShutau